content/realtime/encryption.textile

```diff
@@ -88,44 +92,91 @@ h6(#get-default-params).
+
+h4. Returns
+
+  On success, the method returns a complete "@CipherParams@":#cipher-params object. Failure to create default cipher params will raise an <span lang="ruby,java">"@AblyException@":/realtime/types/#ably-exception </span><span lang="jsall">exception</span>.
+
+h4. Example
+
+```[jsall](code-editor:realtime/channel-encrypted)
+  var cipherParams = Ably.Realtime.Crypto.getDefaultParams(<key>);
+  var channelOpts = { encrypted: true, cipherParams: cipherParams };
+  var channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', channelOpts);
+```
+
+```[ruby]
+  cipher_params = Ably::Util::Crypto.get_default_params(<key>)
+  channel_opts = { encrypted: true, cipher_params: cipher_params }
+  channel = realtime.channels.get('{{RANDOM_CHANNEL_NAME}}', channel_opts)
+```
+
+```[java]
+  CipherParams params = Crypto.getDefaultParams(<key>);
+  ChannelOptions options = new ChannelOptions();
+  options.encrypted = true;
+  options.cipherParams = params;
+  Channel channel = realtime.channels.get("{{RANDOM_CHANNEL_NAME}}", options);
+```
+
+
+h6(#generate-random-key).
+  default:  generateRandomKey
+  ruby:     generate_random_key
+
+bq(definition).
+  default: Crypto.generateRandomKey(Int keyLength?, callback("ErrorInfo":/realtime/types#error-info err, @Buffer@ key))
+  ruby:    byte array Crypto.generate_random_key(Int key_length?)
+  java:    @byte[]@ Crypto.generateRandomKey(Int keyLength?)
 
-This call obtains a "@CipherParams@":#cipher-params object with default values for the implementation, either taking a supplied cipher key or generating a secure random key.
+This call obtains a randomly-generated binary key of the specified key length.
 
 h4. Parameters
 
-- key := an optional <span lang="java">@byte[]@</span><span lang=default>@String@</span> containing the secret key. If not provided, a secure random key is generated
+- keyLength := Optional @Int@ with the length of key to generate. For AES, this should be either 128 or 256. If unspecified, defaults to 256.
 
-- <div lang="jsall">callback</div> := is a function of the form @function(err, cipherParams)@ which is called upon completion
+- <div lang="jsall">callback</div> := is a function of the form @function(err, key)@ which is called upon completion
 
 blang[jsall].
   h4. Callback result
 
-  On successfully creating default cipher params, the callback is called with the default "@CipherParams@":#cipher-params object and @err@ is @null@. On failure to create default cipher params, @err@ contains an "@ErrorInfo@":#error-info object describing the failure reason.
+  On successfully generating a key, the callback is called with that key as a <span lang=javascript>@WordArray@</span><span lang=nodejs>@Buffer@</span>, and @err@ is @null@. On failure to create a key, @err@ contains an "@ErrorInfo@":#error-info object describing the failure reason.
```

Comment:
`WordArray` and `ArrayBuffer` are different datatypes. `ArrayBuffer` is a native javascript binary type (but isn't supported by some browsers); `WordArray` is a CryptoJS custom type. `generateRandomKey` returns a WordArray always, as that's better than returning a different thing depending on what browser you're using. `getDefaultParams` can take a key as [either a WordArray or an ArrayBuffer](https://github.com/ably/docs/pull/89/files#diff-80469b1c5e0dd2e8ff3da0c178748ae7R92).